Add portable mode for Mac

raelyard opened the pull request:
Similar to portable mode for Windows and Linux, but the relative path for the portable .atom home directory has a deeper nesting.

raelyard committed:
Add portable mode for Mac

raelyard committed:
Fix missing parameter for setting portable mode

50Wliu commented on spec/atom-portable-spec.coffee:
```diff
@@ -35,8 +37,9 @@ portableModeCommonPlatformBehavior = (platform) ->
-  portableAtomHomePath = path.join(path.dirname(process.execPath), "..", ".atom")
+fdescribe "Set Portable Mode", ->
+  portableAtomHomePath = path.join(process.resourcesPath, "..", "..", '..', ".atom") if platform is 'darwin'
+  portableAtomHomePath = path.join(path.dirname(process.execPath), "..", ".atom") if not platform is 'darwin'
```
```coffee
if platform is 'darwin'
  portableAtomHomePath = path.join(process.resourcesPath, "..", "..", '..', ".atom")
else
  portableAtomHomePath = path.join(path.dirname(process.execPath), "..", ".atom")
```
?

Also, shouldn't it be `process.platform`?

raelyard committed:
Remove focus specs

that should not have been committed

raelyard committed:
Comment to clarify different path

for portable home for Mac OS

thomasjo commented on src/browser/atom-portable.coffee:
```diff
@@ -4,31 +4,34 @@ ipc = require 'ipc'
 class AtomPortable
-  @getPortableAtomHomePath: ->
+  @getPortableAtomHomePath: (platform) ->
+    # Mac has a deeper path for the executable in the application package, so need a different
```
*OS X* instead of *Mac*. I realize this is super-pedantic, but we really should start standardizing the platform terminology :bowtie:

raelyard replied:
@thomasjo That's a good point and can change this comment, but not very consistent with references elsewhere.  There's more "Mac" than "OS X" in the project (and also some MacOS and OSX).

thomasjo replied:
I'm actually hoping to go through the inconsistencies over the weekend in an attempt to resolve them. So not a major thing if you don't fix this one now, but I'll have to fix it eventually :)

raelyard committed:
Standardize platform reference

in comment

raelyard committed:
Fix broken spec

needing to get the process platform because it's not passed in on the spec for setting portable

kevinsawicki commented on src/browser/main.coffee:
```diff
@@ -64,13 +64,13 @@ setupAtomHome = ({setPortable}) ->
+      AtomPortable.setPortable(process.platform, atomHome)
     catch error
-      console.log("Failed copying portable directory '#{atomHome}' to '#{AtomPortable.getPortableAtomHomePath()}'")
+      console.log("Failed copying portable directory '#{atomHome}' to '#{AtomPortable.getPortableAtomHomePath process.platform}'")
```
Would you mind adding parens around the `getPortableAtomHomePath` call here?

kevinsawicki commented on src/browser/main.coffee:
```diff
@@ -64,13 +64,13 @@ setupAtomHome = ({setPortable}) ->
 
   if AtomPortable.isPortableInstall(process.platform, process.env.ATOM_HOME, atomHome)
-    atomHome = AtomPortable.getPortableAtomHomePath()
+    atomHome = AtomPortable.getPortableAtomHomePath process.platform
```
Would you mind adding parens around the `getPortableAtomHomePath` call here?

raelyard committed:
:art:

raelyard committed:
:art:

raelyard committed:
Limit scope of copying home directory for portable

Setting portable mode happens by copying the atom home from the default
location to a portable locatoin.  When copying the .atom directory, only
copy relevant files and directories and not cache and build and such
because it's unecessary and to prevent problems with trying to unpack asar
archives.

Commit:
Merge branch 'master' into dr-portable-mode-mac

damieng commented:
This was done independently in 1.14.
